packages/wp-preferences/src/database-persistence-layer/index.js

```diff
@@ -0,0 +1,41 @@
+/**
+ * WordPress dependencies
+ */
+import apiFetch from '@wordpress/api-fetch';
+
+export default function createDatabasePersistenceLayer() {
```

Comment:
I kind of don't like `wp-preferences` as a package name. Instead, could `database-persistence-layer` be a package instead?

it could also be `@wordpress/preferences-api-adapter` and be configurable if needed (API path...) and all the gluing happens as an inline script in the server to make it WP specific.

It can even be something the raw preferences package exposes (the configurable adapter) since `api-fetch` is not WordPress specific by default.

Is that too complex?

Reply:
A better name would be good. I also want to think about somewhere to put all the migration code that's in the data package at the moment - https://github.com/WordPress/gutenberg/blob/trunk/packages/data/src/plugins/persistence/index.js#L588, I'm not sure yet whether it's right for this to live alongside the database persistence layer.

I was also considering moving the components out of the preferences package to somewhere else, but that's not entirely necessary.

Using an inline script would also be good. I think it might be best for preferences to do its own preloading and not rely on apiFetch, and this would be a good way to do it.